### ISSUE TITLE
Error when running the Python Sample

### ISSUE THREAD

**Issue by ochowie** [grpc/homebrew-grpc#53]
After installing the package either on Linux or Mac I get the following error when I run a sample from the grpc-common repo.

**\* Linux Version ****
  File "greeter_server.py", line 34, in <module>
    import helloworld_pb2
  File "/home/arthur_zeyda/grpc-common/python/helloworld/helloworld_pb2.py", line 111, in <module>
    from grpc.early_adopter import implementations
  File "/usr/local/lib/python2.7/dist-packages/grpc/early_adopter/implementations.py", line 34, in <module>
    from grpc._adapter import fore as _fore
  File "/usr/local/lib/python2.7/dist-packages/grpc/_adapter/fore.py", line 38, in <module>
    from grpc._adapter import _intermediary_low as _low
  File "/usr/local/lib/python2.7/dist-packages/grpc/_adapter/_intermediary_low.py", line 57, in <module>
    from grpc._adapter import _low
  File "/usr/local/lib/python2.7/dist-packages/grpc/_adapter/_low.py", line 30, in <module>
    from grpc._adapter import _c
ImportError: /usr/local/lib/python2.7/dist-packages/grpc/_adapter/_c.so: undefined symbol: grpc_jwt_credentials_create

**\* Mac Version *****
ImportError: dlopen(/Users/azeyda/Code/grpc/python_virtual_environment/lib/python2.7/site-packages/grpc/_adapter/_c.so, 2): Symbol not found: _gpr_convert_clock_type
  Referenced from: /Users/azeyda/Code/grpc/python_virtual_environment/lib/python2.7/site-packages/grpc/_adapter/_c.so
  Expected in: flat namespace
 in /Users/azeyda/Code/grpc/python_virtual_environment/lib/python2.7/site-packages/grpc/_adapter/_c.so

**Comment by soltanmm** [comment]
Where are you getting/building the gRPC library binaries from? Where are you getting/building the gRPC Python library binary `_c.so` from?

First thoughts:

On July 7 [this commit](https://github.com/grpc/grpc/commit/5400606d2197dff83534bfd7a455dd241f2c8baf) went in via [this PR](https://github.com/grpc/grpc/pull/2323/). That changed the symbols, and [this commit](https://github.com/grpc/grpc/commit/fe4c3f4f1436e2fc2297d5544859ce14e03659bf) changed the Python code on July 22nd via [this PR](https://github.com/grpc/grpc/pull/2617).

I'm not sure, but this might be a problem upstream given that the bottle uses the 0.10.0 release, as that release doesn't appear to contain either of the commits above. I'm not sure, though, without knowing a little more _explicitly_ about how you got your libs.

**Comment by ochowie** [comment]
So I'm building the core gRPC binaries from the main gRPC repo and I assume I'm getting the Python library binary from the bottle(?). As an experiment, I tried following the steps in the GRPC python docker image (which doesn't use brew) and that worked so it seems like your hunch is correct. In fact, on one of my failed iterations, I followed the dockerfile steps for everything but that doesn't build the grpc_python_plugin and when I tried getting that via brew I got the same error as above. However, on the next attempt when i build the grpc_python_plugin from source everything worked correctly. So it seems like there is something missing from the brew installation.

**Comment by soltanmm** [comment]
Yeah, I don't think we have any guarantees that branch master of the gRPC repo will work with anything anywhere at any time. Your best bet's sticking with the release branches (release 0.10.0 should work with the homebrew install).

**Comment by ochowie** [comment]
So you would suggest either building everything from master or grabbing the
release branch? Can I grab the most recent release or does the bottle only
work with specific releases? Also, should the brew install be self
contained or do I still need the core grpc binary?

Thanks,
Arthur

**Comment by soltanmm** [comment]
@tbetbetbe What's the policy?

**Comment by nicolasnoble** [comment]
The policy is that while we're still in alpha, we're very much prone to
break the API and backward compatibility. As we're going to enter Beta,
we'll start to guarantee that the API/ABI will be stable from that point
forward. We might still do API/ABI-breaking changes, but we'll try to
minimize them the best we can, and limit them to crucial updates only. A GA
release will then certify we won't change the API/ABI anymore until the
next major version of grpc.

**Comment by soltanmm** [comment]
I meant more along the lines of: what are our guarantees with respect to this homebrew install in terms of releases and versioning, if any? e.g. the homebrew install references release 0.10.0; do we guarantee that when installed it should work as intended with an install from the release branch of 0.10.0? etc. etc. etc.
